vertx-mssql-client/src/main/java/io/vertx/mssqlclient/MSSQLConnectOptions.java

```diff
@@ -61,6 +61,9 @@ public MSSQLConnectOptions(JsonObject json) {
   
   public MSSQLConnectOptions(SqlConnectOptions other) {
     super(other);
+    if (getProperties().isEmpty()) {
```

Comment:
I think we should not set a default properties mapping here but check those options at connection level.
Can you take a look at https://github.com/eclipse-vertx/vertx-sql-client/pull/636?

If you're fine with that, you could merge it and rebase.